dashboard/src/main/java/com/google/cloud/tools/opensource/cloudbomdashboard/ArtifactMavenData.java

```diff
@@ -220,21 +235,33 @@ private static String getSharedDependenciesVersionFromUrl(String pomUrl) {
       FileUtils.copyInputStreamToFile(input, pomFile);
       MavenXpp3Reader read = new MavenXpp3Reader();
       Model model = read.read(new FileInputStream(pomFile));
-      if (model.getDependencyManagement() == null) {
-        return null;
-      }
-      for (Dependency dep : model.getDependencyManagement().getDependencies()) {
-        if ("com.google.cloud".equals(dep.getGroupId())
-            && "google-cloud-shared-dependencies".equals(dep.getArtifactId())) {
-          if (dep.getVersion().startsWith("${")) {
-            String sharedVersion = dep.getVersion().substring(1).replaceAll("[{}]", "");
-            return model.getProperties().getProperty(sharedVersion);
-          }
-          return dep.getVersion();
-        }
-      }
-
-    } catch (XmlPullParserException | IOException ignored) {
+      ModelBuildingRequest request = new DefaultModelBuildingRequest();
+      request.setRawModel(model);
+      RepositorySystem repositorySystem = RepositoryUtility.newRepositorySystem();
+      RepositorySystemSession session = RepositoryUtility.newSession(repositorySystem);
+
+      // Set model resolver to locate parent POM on Maven Central.
+      RemoteRepository mavenCentral =
+          new RemoteRepository.Builder("central", "default", "https://repo1.maven.org/maven2/")
+              .build();
+      request.setModelResolver(
+          new ProjectModelResolver(
+              session,
+              null,
+              repositorySystem,
+              new DefaultRemoteRepositoryManager(),
+              ImmutableList.of(mavenCentral),
+              null,
+              null));
+
+      // Capture java version from system environment for profile activation.
+      request.setSystemProperties(System.getProperties());
+
+      ModelBuilder builder = new DefaultModelBuilderFactory().newInstance();
+      Model effectiveModel = builder.build(request).getEffectiveModel();
+      return effectiveModel.getProperties().getProperty("google-cloud-shared-dependencies.version");
+    } catch (XmlPullParserException | IOException | ModelBuildingException exception) {
+      LOGGER.log(Level.SEVERE, "Failed to parse contents of POM file: {0}", pomUrl);
```

Comment:
Providing the exception will help future debuggers since it'll show the cause and the stack trace.
```suggestion
      LOGGER.log(Level.SEVERE, exception, () -> "Failed to parse contents of POM file: " + pomUrl);
```

Reply:
Thanks for the suggestion. Accepted.